config/configtls/configtls.go

```diff
@@ -205,6 +194,89 @@ func (c TLSSetting) loadTLSConfig() (*tls.Config, error) {
 	}, nil
 }
 
+func (c TLSSetting) loadCACertPool() (*x509.CertPool, error) {
+	// There is no need to load the System Certs for RootCAs because
+	// if the value is nil, it will default to checking against th System Certs.
+	var err error
+	var certPool *x509.CertPool
+
+	switch {
+	case c.hasCAFile() && c.hasCAPem():
+		return nil, fmt.Errorf("failed to load CA CertPool: CA File and PEM cannot both be provided")
```

Comment:
It would be nice to get a native speaker to check this wording, as it sounds a bit strange to my ears. I would use something like:

> failed to load CA CertPool: provide either a CA file or the PEM-encoded string, but not both

Reply:
I updated some of the verbiage to match the suggested pattern. Let me know if you think more needs to be changed. It's no big deal on my end.